pkg/rancher-desktop/assets/lima-config.yaml

```diff
@@ -23,23 +23,43 @@ provision:
     mkdir -p /bootfs
     mount --bind / /bootfs
     # /bootfs/etc is empty on first boot because it has been moved to /mnt/data/etc by lima
-    if [ -f /bootfs/etc/os-release ] && ! diff -q /etc/os-release /bootfs/etc/os-release; then
-      cp /etc/machine-id /bootfs/etc
-      cp /etc/ssh/ssh_host* /bootfs/etc/ssh/
-      mkdir -p /etc/docker /etc/rancher
-      cp -pr /etc/docker /bootfs/etc
-      cp -pr /etc/rancher /bootfs/etc
+    if [ -f /bootfs/etc/os-release ]; then
+      # Alpine turned /etc/os-release into a symlink; we dereference it again. If we still have a symlink
+      # here, then this is an upgrade from an older version and needs to go through the migration.
+      if [ -L /etc/os-release ] || ! diff -q /etc/os-release /bootfs/etc/os-release; then
+        # When we are upgrading from an ISO that didn't have iproute2 installed, we will install it during boot.
+        # But Lima will restore the old /etc/apk/world file and run `apk fix --no-network` to restore packages from
+        # cache that were installed manually. This will however uninstall iproute2 again because it is not listed
+        # in the old world file. We need to install it once more from the boot media to update the world file.
+        if ! apk info -e iproute2 >/dev/null 2>&1; then
+          apk add iproute2 --no-network --keys-dir /bootfs/etc/apk/keys --repositories-file /bootfs/etc/apk/repositories
+        fi
 
-      rm -rf /mnt/data/etc.prev
-      mkdir /mnt/data/etc.prev
-      mv /etc/* /mnt/data/etc.prev
-      mv /bootfs/etc/* /etc
+        cp -L /etc/os-release /etc/os-release.tmp
+        mv /etc/os-release.tmp /etc/os-release
```

Comment:
Turns out (at least in busybox) `cp -L file file` can be used to dereference the symlink. (It only works when it's a symlink; when it's a regular file it produces an error.)

I guess this eventually ends up in `/mnt/data/etc.prev` before we do another copy on line 50/52?